src/WinRT.Runtime/Marshalers.cs

```diff
@@ -1195,13 +1197,23 @@ static Marshaler()
                     FromManaged = (T value) => value;
                     DisposeMarshaler = (object box) => { };
                     DisposeAbi = (object box) => { };
-                    CreateMarshalerArray = (T[] array) => MarshalBlittable<T>.CreateMarshalerArray(array);
-                    GetAbiArray = (object box) => MarshalBlittable<T>.GetAbiArray(box);
-                    FromAbiArray = (object box) => MarshalBlittable<T>.FromAbiArray(box);
-                    FromManagedArray = (T[] array) => MarshalBlittable<T>.FromManagedArray(array);
-                    CopyManagedArray = (T[] array, IntPtr data) => MarshalBlittable<T>.CopyManagedArray(array, data);
-                    DisposeMarshalerArray = (object box) => MarshalBlittable<T>.DisposeMarshalerArray(box);
-                    DisposeAbiArray = (object box) => MarshalBlittable<T>.DisposeAbiArray(box);
+                    if (type.IsEnum)
+                    {
+                        // For marshaling non-blittable enum arrays via MarshalNonBlittable
+                        unsafe void CopyEnum(object value, IntPtr dest) => *(int*)dest.ToPointer() = (int)Convert.ChangeType(value, typeof(int));

```

Comment:
Will all enum values that we get be convertible to int and back without issues?  For instance max value of uint32 for flags enum?

Reply:
yes, good call - flags require special treatment